core/state-machine/src/basic.rs

```diff
@@ -172,6 +172,19 @@ impl<H: Hasher> Externalities<H> for BasicExternalities where H::Out: Ord {
 		warn!("Call to non-existent out offchain externalities set.");
 		None
 	}
+
+	fn start_transaction(&mut self) {
+		warn!("No support for transaction");
+	}
+
+	fn discard_transaction(&mut self) {
+		panic!("No support for transaction");
```

Comment:
Can you explain why these are noops/panics?

Reply:
BasicExternalities is currently only use for genesis init, so this makes failing transaction panic during genesis build. That is not ideal, especially considering BasicExteranalities could be use in different context.
So I see two way of doing things:

- document basic externalities limitation properly (no transaction, no offchain ext) and document genesis initialisation limitation.
- implement transaction by switching back to internal overlay, this is quite straight forward. May be a bit heavy, but considering that it is use for genesis only it may not be such an issue.

cc/ @thiolliere since I know you know this code.

Reply:
I don't see any other use than genesis for now.

I'm also mixed: providing such features for building genesis storage is of no interest. but having an API that have panicking function because you're not suppose to use it is also not very good.

Also I see some assert in the building of the genesis in srml-staking, maybe no panic policy is not same importance for genesis build ?

Reply:
maybe it is better to have no-op and a warning like it does for offchain workers

Reply:
> Also I see some assert in the building of the genesis in srml-staking, maybe no panic policy is not same importance for genesis build ?

I would assume so.  No untrusted inputs go into building genesis, and it is much better to catch problems before a chain is launched.